### PR TITLE
Use buffered sender by default instead of the sync sender

### LINKED ISSUE #117: Make an async sender the default
- Use one of the queue based senders as the default sender instead of the current sync sender
- Document that the default is async and what the queueing strategy is
- Document how to replace the queue and the whole sender

### PR DESCRIPTION
**Pull request by basoko**
Fix #117

### CODE REVIEW

**Comment by rokob** [comment]
So in other notifiers where we have an async send mechanism we also expose a function `wait` which when called stops the queue from accepting items and blocks until all of the items have been sent. I think that is beyond the scope of this PR but I just wanted to bring it up so I remember to add it later.

**Comment by basoko** [comment]
No sure exactly the use case for that, but it seems that we could have the same behavior with a bounded Queue and some extra flags in the buffered sender to determine this behaviour. Now when the queue is full it just discard the element and notify about it through the `SenderListener`, so we could make it to block until there is more space.

**Comment by rokob** [comment]
The use case is to delay shutdown until items have been processed; i.e. the use of sleep in your test would be replaced with a call to wait. But I forgot about the `SenderListener` so one could build that themselves.

**Comment by basoko** [comment]
Ok! I see it now! Thanks for the explanation!

File: README.md [hunk 1]
```diff
@@ -185,6 +185,15 @@ compile('com.rollbar:rollbar-java:1.0.1')
 compile('com.rollbar:rollbar-android:1.0.1@aar')
 ```
 
+## How payloads are sent
+
+The actual notifier configuration builds a notifier that uses a BufferedSender to send the items
+to Rollbar. That sender is built using an unbound memory queue and a scheduled thread to send
+the events from the queue. 
+The queue and as well the frequency of the scheduled thread can be customize
```
Comment (mbryzek) [inline README.md]: customize => customized

Comment (mbryzek) [reply README.md]: `The queue and` => the `and` should be removed

Comment (basoko) [reply README.md]: Thanks!

**Pull request merged**